docs(mssql): describe MSSQL issue with negative certificates (#3417)
testcontainers/testcontainers-go: docs/modules/mssql.md

go get github.com/testcontainers/testcontainers-go/modules/mssql
```

> !!!info
>     In order to use this module, you must set the `GODEBUG=x509negativeserial=1` environment variable. See [https://github.com/microsoft/mssql-docker/issues/895](https://github.com/microsoft/mssql-docker/issues/895) for more details.
> 

Author: ```suggestion
!!!info
    To use this module with Go 1.23+, set `GODEBUG=x509negativeserial=1`. See the related issue in the [mssql-docker repository](https://github.com/microsoft/mssql-docker/issues/895) for details.
    
    ```shell
    # append to any existing GODEBUG flags instead of overwriting
    export GODEBUG="${GODEBUG:+$GODEBUG,}x509negativeserial=1"
    ```
```

go get github.com/testcontainers/testcontainers-go/modules/mssql
```

!!!info

>     In order to use this module, you must set the `GODEBUG=x509negativeserial=1` environment variable. See https://github.com/microsoft/mssql-docker/issues/895 for more details.

Author: ```suggestion
    In order to use this module, you must set the `GODEBUG=x509negativeserial=1` environment variable. See [https://github.com/microsoft/mssql-docker/issues/895](https://github.com/microsoft/mssql-docker/issues/895) for more details.
```

This is happening because:
- The MSSQL Docker image uses a self-signed certificate with a negative serial number
- Go 1.23+ has stricter certificate validation that rejects certificates with negative serial numbers by default
- The `x509negativeserial=1` flag tells Go to accept certificates with negative serial numbers

The error you're seeing is a security feature in Go 1.23+ that was introduced to prevent potential certificate-related attacks. The MSSQL Docker image hasn't been updated to use certificates with positive serial numbers yet, which is why we need to use this workaround.

!!!info

>     This is fixed in SQL2019 CU32 and SQL2022 CU18:
> https://learn.microsoft.com/en-us/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate18#3867855

Author: ```suggestion
    This is fixed in SQL2019 CU32 and SQL2022 CU18: [https://learn.microsoft.com/en-us/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate18#3867855](https://learn.microsoft.com/en-us/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate18#3867855)
```
